File: 9-palindrome-number/palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        int y =x;
        int result = 0;
        while(y>0){
            int digit = y%10;
            result = (result*10) + digit;
            y =y/10; 

        }
        if(result==x) return true;


        return false;
        
    }
}

File: 9-palindrome-number/palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        int num =x;
        long long rev =0;
       if (x < 0 || (x % 10 == 0 && x != 0))
            return false;
        while(num!=0){
            int digit = num%10;
            rev = rev*10+digit;
            num = num/10;
        }

        if(x==rev) return true;

        return  false;
        
    }
};